internal/pkg/es/result.go

```diff
@@ -54,8 +57,63 @@ type HitsT struct {
 	MaxScore *float64 `json:"max_score"`
 }
 
+type Bucket struct {
+	Key          string           `json:"key"`
+	DocCount     int64            `json:"doc_count"`
+	Aggregations map[string]HitsT `json:"-"`
+}
+
+type _bucket Bucket
```

Comment:
why?

Reply:
This is needed because otherwise an infinite loop will occur on the unmarshal because `Bucket` as the `UnmarshalJSON` function defined so it implements the interface.

By using a type alias it ensures that it will not recall the same method over and over again. So this must be used this way.